tests/Drupal/Semantics/FunctionTUnitTest.inc

```diff
@@ -34,3 +34,8 @@ class Test {
   }
 
 }
+
+// Using constants in t() should be allowed.
+$x = t(FOO);
+$x = t(Test::FOO);
+$x = t(self::FOO);
```

Comment:
Ah sorry, now I understand your previous changes.

I think the self test case is not correct here, this should be called within a class. Self outside of classes is invalid, does this even parse as T_SELF? Best move this test in the class above in a new test method and refer to the class constant there.